src/sparse/impl/KokkosSparse_spmv_bsrmatrix_spec.hpp

```diff
@@ -50,7 +50,8 @@
 
 #include "KokkosSparse_BsrMatrix.hpp"
 #include "KokkosKernels_Controls.hpp"
-#if !defined(KOKKOSKERNELS_ETI_ONLY) || KOKKOSKERNELS_IMPL_COMPILE_LIBRARY
+#if !defined(KOKKOSKERNELS_ETI_ONLY) || \
+    KOKKOSKERNELS_IMPL_COMPILE_LIBRARY && defined(__CUDA_ARCH__)
```

Comment:
I think there should be parentheses around the ``||`` expression, since we want "old  condition && compiling for cuda device"

Reply:
yeah, good point, I'll do that